azurerm/internal/services/iothub/iothub_resource.go

```diff
@@ -239,30 +240,37 @@ func resourceArmIotHub() *schema.Resource {
 							},
 							Sensitive: true,
 						},
+
+						"endpoint_resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
```

Comment:
I don't think we need to prefex this with endpoint as it is already in the endpoint block?

Reply:
and this is a breaking change as it makes this property required? could we not used shared schema and write an optional one by hand?

Reply:
updated